src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -126,7 +126,7 @@ public ITempDataDictionary TempData
         }
 
         /// <inheritdoc />
-        public Action<TextWriter> RenderBodyDelegate { get; set; }
+        public Func<TextWriter, Task> RenderBodyDelegateAsync { get; set; }
```

Comment:
Might end up cleaning this up as part of https://github.com/aspnet/Mvc/issues/3030